sds_in_a_box/SDSCode/opensearch_utils/query.py

```diff
@@ -0,0 +1,101 @@
+import json
+
+class Query:
+    """
+    Query class to represent an AWS OpenSearch query domain-specific language (DSL),
+    providing a range of search options. Query objects will be used by the Client
+    search method and used in OpenSearch searches.
+
+    ...
+
+    Attributes
+    ----------
+    query: dict
+        dictionary containing the query field (dict key) and the query text 
+        (dict value).
+    size: int, optional
+        number of results to return. default is 10.
+    
+
+    Methods
+    -------
+    query_dsl: str
+        returns the query in the OpenSearch Query DSL format
+    size: int
+        returns the number of results the query is allowed to return in the search.
+    """
+
+    def __init__(self, query_params, size=10):
+        self.query_params = query_params
+        self.query_dsl_formatted = self.__build_query_dsl(query_params)
+        self.query_size = size 
+
+    def query_dsl(self):
+        """Returns the query parameters in the AWS OpenSearch Query DSL format"""
+        return json.dumps(self.query_dsl_formatted)
+
+    def size(self):
+        """Returns the number of results the query is allowed to return in the search"""
+        return self.query_size
+
+    def __build_query_dsl(self, query_params):
+        """
+        Builds a Query DSL using a dictionary with field:value pairings.
+
+        Parameters
+        ----------
+        query_params: dict
+            dictionary containing field:value search parameters.
+
+        """
+        # define the query structure
+        query = {"query": {"bool":{}}}
```

Comment:
This structure is to match the Query DSL format: https://opensearch.org/docs/latest/opensearch/query-dsl/bool/

I think opensearch needs to see that query key to know that a query is contained in it because you can send more complicated json requests with your search. For example, you could send a query key along with a sort key that will query and then sort the results that are returned:

{
  "query": {
    "term": {
      "play_name": {
        "value": "Henry IV"
      }
    }
  },
  "sort": [
    {
      "line_id": {
        "order": "desc"
      }
    }
  ]
}

https://opensearch.org/docs/2.4/opensearch/search/sort

It looks a little weird since we're just sending queries right now, but I think that key is necessary.